### PR TITLE
Do not populate transaction cache when confirming from link.

### PR DESCRIPTION
**Pull request by evoskuil**

### CODE REVIEW

**Comment by kulpreet** [comment]
Will this will result in `cache_.add` _not_ being called on `transaction_database::unconfirm`?

Is that a non-concern?

**Comment by evoskuil** [comment]
@kulpreet yes, real issue. Nice catch.

File: src/databases/transaction_database.cpp [hunk 1]
```diff
@@ -421,45 +424,60 @@ bool transaction_database::candidize(link_type link, bool candidate)
...
+// This also does not confirm unconfirmed cached outputs (should not exist).
+// This also does not remove cached items on unconfirmation (should not call).
+bool transaction_database::confirm(file_offset link, size_t height,
+    uint32_t median_time_past, size_t position)
 {
-    uint32_t position = 0;
-    for (const auto& tx: transactions)
-        if (!confirm(tx.metadata.link, height, median_time_past, position++))
+    // Avoid population of prevouts for coinbase (confirmation optimization).
+    if (position != 0)
```
Comment (kulpreet) [inline src/databases/transaction_database.cpp]: @evoskuil this has an implication on the on the `transaction_result::is_spent` method. We will need to skip the spentness check on the 0th output.

I came across it when trying to bring back the commented out test for confirming a list of transactions; which has now changed to confirming transactions in a block.

Do we need this test? I ask, because in the confirmed_spend mehtod we anyway check if point is null.

Comment (kulpreet) [reply src/databases/transaction_database.cpp]: Actually, ignore all of the above. It is the tests that needed to be fixed a bit. I'll send a PR for your fork with the tests fixed.

**Comment by evoskuil** [comment]
Yeah, the reason to use block here is so we can infer Coinbase from order. Previously that was not technically possible.

**Comment by evoskuil** [comment]
@kulpreet thanks for the quick review.
